cluster/hosts.go

```diff
@@ -54,14 +56,37 @@ func (c *Cluster) TunnelHosts(ctx context.Context, flags ExternalFlags) error {
 		return err
 	}
 	for _, host := range c.InactiveHosts {
-		log.Warnf(ctx, "Removing host [%s] from node lists", host.Address)
-		c.EtcdHosts = removeFromHosts(host, c.EtcdHosts)
-		c.ControlPlaneHosts = removeFromHosts(host, c.ControlPlaneHosts)
-		c.WorkerHosts = removeFromHosts(host, c.WorkerHosts)
-		c.RancherKubernetesEngineConfig.Nodes = removeFromRKENodes(host.RKEConfigNode, c.RancherKubernetesEngineConfig.Nodes)
+		c.removeHostFromNodeList(ctx, host)
 	}
 	return ValidateHostCount(c)
+}
+
+func (c *Cluster) RemoveHostsLabeledToIgnoreUpgrade(ctx context.Context) {
+	kubeClient, err := k8s.NewClient(c.LocalKubeConfigPath, c.K8sWrapTransport)
+	nodes, err := kubeClient.CoreV1().Nodes().List(metav1.ListOptions{})
```

Comment:
Should we use GetNode here with retries to make sure we did our best to get the nodes?

Reply:
might be excessive to call GetNode on each host with retries, but I'll add retry logic around this List call